src/System.Runtime/tests/System/Text/StringBuilderTests.netcoreapp.cs

```diff
@@ -199,5 +199,38 @@ public static void Insert_CharSpan_Invalid()
             AssertExtensions.Throws<ArgumentOutOfRangeException>("index", () => builder.Insert(builder.Length + 1, new ReadOnlySpan<char>(new char[0]))); // Index > builder.Length
             AssertExtensions.Throws<ArgumentOutOfRangeException>("requiredLength", () => builder.Insert(builder.Length, new ReadOnlySpan<char>(new char[1]))); // New length > builder.MaxCapacity
         }
+
+        public static IEnumerable<object[]> Equals_String_TestData()
+        {
+            var mediumString = new string('a', 30);
+            var largeString = new string('a', 1000);
+            var extraLargeString = new string('a', 41000); // 8000 is the maximum chunk size
+
+            var sb1 = new StringBuilder("Hello");
+            var sb2 = new StringBuilder(20).Append(mediumString);
+            var sb3 = new StringBuilder(20).Append(largeString);
+            var sb4 = new StringBuilder(20).Append(extraLargeString);
+
+            yield return new object[] { sb1, "Hello", true };
+            yield return new object[] { sb1, "Hel", false };
+            yield return new object[] { sb1, "Hellz", false };
+            yield return new object[] { sb1, "Hellz0", false };
```

Comment:
Not sure what is the value of this test case, given that we have the one above.

Reply:
Actually thats a typo. I intend  to write helloz to check the behavior when length of string is greater than the stringBuilder. (Will make the change)